### PR TITLE
Fix evbuffer_file_segment_new 64-bit support on Win32

### PR DESCRIPTION
**Pull request by Coeur**
This fixes the problematic `#define fstat _fstat` which would only support files up to 2 GB.

Also refactored it as `evutil_fd_filesize` to avoid exposing `stat` when not necessary.

### CODE REVIEW

File: evutil.c [hunk 1]
```diff
@@ -193,12 +202,13 @@ evutil_read_file_(const char *filename, char **content_out, size_t *len_out,
 	fd = evutil_open_closeonexec_(filename, mode, 0);
 	if (fd < 0)
 		return -1;
-	if (fstat(fd, &st) || st.st_size < 0 ||
-	    st.st_size > EV_SSIZE_MAX-1 ) {
+	length = evutil_fd_filesize(fd);
+	if (length < 0 ||
+        length > EV_SSIZE_MAX-1 ) {
```
Comment (azat) [inline evutil.c:207]:
```suggestion
	if (length < 0 || length > EV_SSIZE_MAX-1 ) {
```

**Review by azat — COMMENTED** [review]
Please fix the build

```
home/runner/work/libevent/libevent/buffer.c:3022:14: error: comparison of integer expressions of different signedness: ‘int64_t’ {aka ‘long int’} and ‘long unsigned int’ [-Werror=sign-compare]
 3022 |      (length > EVBUFFER_CHAIN_MAX) ||
```

File: evutil.c (hunk 1 above)
Comment (Coeur) [reply evutil.c]: 56e17b3897e229503864ecb86bd5a88e1b17157d

File: event.c [hunk 2]
```diff
@@ -2578,7 +2578,7 @@ evthread_notify_base_default(struct event_base *base)
 #ifdef _WIN32
 	r = send(base->th_notify_fd[1], buf, 1, 0);
 #else
-	r = write(base->th_notify_fd[1], buf, 1);
```
Comment (azat) [inline event.c:2581]: Why do you did cast to `int`, instead of converting `r` to `ssize_t`? And more importantly, how this is related to the initial problem? Let's revert this change

File: evutil.c [hunk 3]
```diff
@@ -160,6 +161,14 @@ evutil_open_closeonexec_(const char *pathname, int flags, unsigned mode)
 	return fd;
 }
 
+ev_off_t evutil_fd_filesize(evutil_socket_t fd)
+{
+    struct stat st;
```
Comment (azat) [inline evutil.c:166]: This file uses tabs not spaces

File: event.c (hunk 2 above)
Comment (Coeur) [reply event.c:2581]: Highest value possible would be `1`, so no need for ssize_t?

**Review by azat — COMMENTED** [review]
Apart from this minor changes looks good

File: event.c (hunk 2 above)
Comment (azat) [reply event.c:2581]: This is just general sense, function return `ssize_t` let just use the same type. But anyway this change is not related for the this PR and should be removed.

File: evutil.c (hunk 3 above)
Comment (Coeur) [reply evutil.c]: b1f5c0491a6ad3a52c3309f5026d338d03d45e68
